packages/adblocker-electron-preload/preload.ts

```diff
@@ -47,7 +50,22 @@ if (window === window.top && window.location.href.startsWith('devtools://') ===
       },
     );
 
-    getCosmeticsFilters({ lifecycle: 'start', ids: [], classes: [], hrefs: [] });
+    const scripts = getCosmeticsFiltersFirst();
+    if (scripts) {
+      const elems: HTMLScriptElement[] = [];
+      try {
+        scripts.forEach((script) => {
+          const e = document.createElement('script');
+          e.appendChild(document.createTextNode(script));
+          (document.head || document.documentElement || document).appendChild(e);
+          elems.push(e);
+        });
+      } catch (ex) {}
+      elems.forEach((removeIt) => {
+        removeIt.remove();
+        removeIt.textContent = '';
+      });
+    }
```

Comment:
Could we reuse the following helpers here?

1. Wrap into an auto-removed script: https://github.com/ghostery/adblocker/blob/master/packages/adblocker-content/adblocker.ts#L236
2. Inject the script: https://github.com/ghostery/adblocker/blob/master/packages/adblocker-content/adblocker.ts#L255

The `1.` above can also be done from the Electron main process before sending the scriptlets to inject.

Reply:
Strictly speaking, the injectScript method of adblocker-content may not work. This is because, for the sake of injecting scripts as quickly as possible, const parent = doc.head || doc.documentElement needs to be changed to const parent = doc.head || doc.documentElement || doc. Also, executing webContent.executeJavascript from Electron's main process requires waiting for the did-stop-loading event, so placing the script in the preload should actually allow for faster code injection. See: https://github.com/electron/electron/blob/main/lib/browser/api/web-contents.ts#L172